crates/indexer/src/accountsdb/storefronts_v2.rs

```diff
@@ -0,0 +1,205 @@
+use indexer_core::{
+    db::{
+        insert_into,
+        models::{
+            SettingsUriJson, StoreConfig as DbStoreConfig, Storefrontv2,
+            WhitelistedCreator as DbWhitelistedCreator,
+        },
+        tables::{settings_uri_jsons, storefrontsv2, storefrontsv2_configs, whitelisted_creators},
+    },
+    hash::HashMap,
+    prelude::*,
+    pubkeys::find_store_config,
+};
+use mpl_metaplex::{
+    state::{
+        Key, Store, StoreConfig, WhitelistedCreator, MAX_STORE_CONFIG_V1_SIZE, MAX_STORE_SIZE,
+        MAX_WHITELISTED_CREATOR_SIZE,
+    },
+    utils::try_from_slice_checked,
+};
+use reqwest::Url;
+use serde::{Deserialize, Serialize};
+
+use crate::{prelude::*, Client};
+const STORE: u8 = Key::StoreV1 as u8;
+const STORE_CONFIG: u8 = Key::StoreConfigV1 as u8;
+const WHITELISTED_CREATOR: u8 = Key::WhitelistedCreatorV1 as u8;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Creator {
+    pub creator_address: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Metadata {
+    pub name: String,
+    pub description: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Address {
+    pub owner: String,
+    pub auction_house: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct SettingUri {
+    meta: Metadata,
+    logo_url: String,
+    banner_url: String,
+    subdomain: String,
+    addresses: Address,
+    creators: Option<Vec<Creator>>,
+    #[serde(flatten)]
+    extra: HashMap<String, serde_json::Value>,
+}
+
+/// process the settings_uri URL
+async fn process_settings_uri(client: &Client, uri: String, pda: String) -> Result<()> {
+    println!("process_settings_uri called!");
+    let url = Url::parse(&uri).context("Failed to parse settings_uri URL!")?;
+    let http_client = reqwest::Client::new();
+    // get the data and parse it into the SettingUri struct
+    let json = http_client
+        .get(url.clone())
+        .send()
+        .await
+        .context("Metadata JSON request failed")?
+        .json::<SettingUri>()
+        .await
+        .context("Failed to parse metadata JSON")?;
+    let row = SettingsUriJson {
+        store_config_pda: Owned(pda),
+        name: Owned(json.meta.name),
+        description: Owned(json.meta.description),
+        logo_url: Owned(json.logo_url),
+        banner_url: Owned(json.banner_url),
+        subdomain: Owned(json.subdomain),
+        owner_address: Owned(json.addresses.owner),
+        auction_house_address: Owned(json.addresses.auction_house),
```

Comment:
```suggestion
        owner_address: Owned(json.address.owner),
        auction_house_address: Owned(json.address.auction_house),
```

https://ikh2nuackoosoqgenrvxqvbaw2r2fd36h2upvfhhup33ref7knya.arweave.net/Qo-m0AJTnSdAxGxreFQgtqOij34-qPqU56P3uJC_U3A